hera_cal/abscal.py

```diff
@@ -185,12 +182,20 @@ def __init__(self, model, data, wgts=None, antpos=None, freqs=None,
         # setup ants
         self.ants = np.unique(np.concatenate(map(lambda k: k[:2], self.keys)))
         self.Nants = len(self.ants)
+        if refant is None:
+            refant = self.keys[0][0]
+            print "using {} for reference antenna".format(refant)
+        else:
+            assert refant in self.ants, "refant {} not found in self.ants".format(refant)
+        self.refant = refant
 
         # setup antenna positions
         self.antpos = antpos
         self.antpos_arr = None
         self.bls = None
         if self.antpos is not None:
+            # center antpos about reference antenna
+            self.antpos = odict(map(lambda k: (k, antpos[k] - antpos[self.refant]), antpos.keys()))
```

Comment:
Is this actually necessary? I thought absolute antenna position didn't matter, just relative antenna position.

Reply:
yes. this is how you enact a reference antenna when solving for a phase slope, rather than a phase